src/cli/internal/studioctlserver/client.go

```
@@ -187,6 +187,20 @@ type AppUpgradeResult struct {
 	ExitCode int              `json:"exitCode"`
 }
 
+// The upgrade exit codes studioctl-server reports, mirroring the Exit* constants in V8Tov9Upgrade.cs.
+const (
+	AppUpgradeExitSuccess            = 0 // Everything applied cleanly.
+	AppUpgradeExitError              = 1 // The upgrade failed.
+	AppUpgradeExitUnsupportedVersion = 2 // The app is not on a version this upgrade can start from.
+	AppUpgradeExitManualRequired     = 3 // Applied, but something is left for the user to finish by hand.
+)
+
+// Indicates whether the upgrade hit a hard error. Any non-zero code that is not the manual-action signal counts as failure
```

golangci-lint (2.12.2)

[warning] exported: comment on exported method AppUpgradeResult.Failed should be of the form "Failed ..."

(revive)